server/leader.go

```diff
@@ -191,6 +185,18 @@ func (s *Server) campaignLeader() error {
 		return errors.Trace(err)
 	}
 
+	s.enableLeader(true)
+	defer func() {
+		s.enableLeader(false)
+		// Reset connections and cluster.
+		s.closeAllConnections()
+		s.cluster.stop()
+		// Clear ts.
+		s.ts.Store(&atomicObject{
```

Comment:
It will panic: https://github.com/golang/go/blob/master/src/sync/atomic/value.go#L50
Maybe we can `s.ts = atomic.Value{}`.